### ISSUE TITLE
nn.Parameter for bias term

### ISSUE THREAD

**Issue by gracikk-ds** [ambisinister/lossfreebalance#1]
Hello,

Thank you for your great work!

I wanted to clarify why we need to use nn.Parameter for biases if we have a rule-based approach for updating them:
`self.expert_biases = nn.Parameter(torch.zeros(num_experts))`

Thanks!

**Comment by WKQ9411** [comment]
I believe this might be because although we update the bias based on rules, once the model is trained, this bias will be fixed as part of the model parameters, hence the need to initialize it using `nn.Parameter`. Additionally, in DeepSeek MoE, because this bias is involved in the `topk` function, it is non-differentiable, meaning the bias does not participate in backpropagation and is only updated through rules. This is my personal understanding, and I am not sure if it is accurate. Maybe we can discuss and exchange ideas further.

**Comment by localhost433** [comment]
Maybe write `self.register_buffer("b", torch.zeros(num_experts))` and then via `torch.no_grad()`? Because it's not something that's getting updated by the optimizer, so if it is `nn.Parameter` the gradients would show up I guess.
